samples/boards/nrf/nrfx/src/main.c

```diff
@@ -41,76 +44,101 @@ void main(void)
+		return;
+	}
 
 	/* Initialize input pin to generate event on high to low transition
 	 * (falling edge) and call button_handler()
 	 */
-	err = nrfx_gpiote_in_init(INPUT_PIN, &in_config, button_handler);
+	static const nrfx_gpiote_input_config_t input_config = {
+		.pull = NRF_GPIO_PIN_PULLUP,
+	};
+	const nrfx_gpiote_trigger_config_t trigger_config = {
+		.trigger = NRFX_GPIOTE_TRIGGER_HITOLO,
+		.p_in_channel = &in_channel,
+	};
+	static const nrfx_gpiote_handler_config_t handler_config = {
+		.handler = button_handler,
+	};
+	err = nrfx_gpiote_input_configure(INPUT_PIN,
+					  &input_config,
+					  &trigger_config,
+					  &handler_config);
 	if (err != NRFX_SUCCESS) {
-		LOG_ERR("nrfx_gpiote_in_init error: %08x", err);
+		LOG_ERR("nrfx_gpiote_input_configure error: 0x%08X", err);
 		return;
 	}
 
-	nrfx_gpiote_out_config_t const out_config = {
-		.action = NRF_GPIOTE_POLARITY_TOGGLE,
-		.init_state = 1,
-		.task_pin = true,
-	};
-
 	/* Initialize output pin. SET task will turn the LED on,
 	 * CLR will turn it off and OUT will toggle it.
 	 */
-	err = nrfx_gpiote_out_init(OUTPUT_PIN, &out_config);
+	static const nrfx_gpiote_output_config_t output_config = {
+		.drive = NRF_GPIO_PIN_S0S1,
+		.input_connect = NRF_GPIO_PIN_INPUT_DISCONNECT,
+		.pull = NRF_GPIO_PIN_NOPULL,
+	};
+	const nrfx_gpiote_task_config_t task_config = {
+		.task_ch = out_channel,
+		.polarity = NRF_GPIOTE_POLARITY_TOGGLE,
+		.init_val = 1,
+	};
+	err = nrfx_gpiote_output_configure(OUTPUT_PIN,
+					   &output_config,
+					   &task_config);
 	if (err != NRFX_SUCCESS) {
-		LOG_ERR("nrfx_gpiote_out_init error: %08x", err);
+		LOG_ERR("nrfx_gpiote_output_configure error: 0x%08X", err);
 		return;
 	}
 
-	nrfx_gpiote_in_event_enable(INPUT_PIN, true);
+	nrfx_gpiote_trigger_enable(INPUT_PIN, true);
 	nrfx_gpiote_out_task_enable(OUTPUT_PIN);
 
 	LOG_INF("nrfx_gpiote initialized");
 
 	/* Allocate a (D)PPI channel. */
 #if defined(DPPI_PRESENT)
-	uint8_t channel;
-	err = nrfx_dppi_channel_alloc(&channel);
+	uint8_t ppi_channel;
+
+	err = nrfx_dppi_channel_alloc(&ppi_channel);
 #else
-	nrf_ppi_channel_t channel;
-	err = nrfx_ppi_channel_alloc(&channel);
+	nrf_ppi_channel_t ppi_channel;
+
+	err = nrfx_ppi_channel_alloc(&ppi_channel);
 #endif
 	if (err != NRFX_SUCCESS) {
-		LOG_ERR("(D)PPI channel allocation error: %08x", err);
+		LOG_ERR("(D)PPI channel allocation error: 0x%08X", err);
 		return;
 	}
 
 	/* Configure endpoints of the channel so that the input pin event is
 	 * connected with the output pin OUT task. This means that each time
 	 * the button is pressed, the LED pin will be toggled.
 	 */
-	nrfx_gppi_channel_endpoints_setup(channel,
+	nrfx_gppi_channel_endpoints_setup(ppi_channel,
 		nrfx_gpiote_in_event_addr_get(INPUT_PIN),
 		nrfx_gpiote_out_task_addr_get(OUTPUT_PIN));
 
 	/* Enable (D)PPI channel. */
 #if defined(DPPI_PRESENT)
-	err = nrfx_dppi_channel_enable(channel);
+	err = nrfx_dppi_channel_enable(ppi_channel);
```

Comment:
can't we switch to `nrfx_gppi_channels_enable` on the occasion? Or this is out of scope of this PR?

Reply:
Good point. I'll change this.